src/System.Text.Json/tests/JsonDocumentTests.cs

```diff
@@ -3691,4 +3745,25 @@ private static string GetCompactJson(TestCaseType testCaseType, string jsonStrin
             return s_compactJson[testCaseType] = existing;
         }
     }
+
+    public class ThrowOnReadStream : MemoryStream
+    {
+        public ThrowOnReadStream(byte[] bytes) : base(bytes)
+        {
+        }
+
+        public override int Read(byte[] buffer, int offset, int count)
+        {
+            throw new EndOfStreamException();
+        }
+    }
+
+    public class ThrowOnArrayPoolRentStep : MemoryStream
```

Comment:
`ThrowOnCanSeekStream` seems better.  ArrayPoolRent is what you're building coverage for right now, but this type might be useful for other tests later.

Reply:
Good point